src/TestSuite/Schema/SchemaCleaner.php

```diff
@@ -0,0 +1,140 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @since         4.3.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\TestSuite\Schema;
+
+use Cake\Console\ConsoleIo;
+use Cake\Database\Schema\BaseSchema;
+use Cake\Database\Schema\CollectionInterface;
+use Cake\Database\Schema\TableSchema;
+use Cake\Datasource\ConnectionInterface;
+use Cake\Datasource\ConnectionManager;
+
+/**
+ * This class will help dropping and truncating all tables of a given connection
+ */
+class SchemaCleaner
+{
+    /**
+     * @var \Cake\Console\ConsoleIo|null
+     */
+    protected $io;
+
+    /**
+     * SchemaCleaner constructor.
+     *
+     * @param \Cake\Console\ConsoleIo|null $io Outputs if provided.
+     */
+    public function __construct(?ConsoleIo $io = null)
+    {
+        $this->io = $io;
+    }
+
+    /**
+     * Drop all tables of the provided connection.
+     *
+     * @param string $connectionName Name of the connection.
+     * @return void
+     * @throws \Exception if the dropping failed.
+     */
+    public function drop(string $connectionName)
```

Comment:
```suggestion
    public function dropTables(string $connectionName)
```

Should these names indicate that they will operate on all tables in the connection? I can make that change if you're ok with it.